File: turunan/part_1/turunan_pertama.cpp
#include <iostream>
#include <functional>

double posisi(double t) {
  return t * t * t - 6 * t * t * 9 * t;
}

double fungsi_turunan(double t) {
  return 3 * t * t - 12 * t + 9;
}

double hitung(std::function<double(double)> fungsi_turunan, double waktu) {
  return fungsi_turunan(waktu);
}

int main() {
  const double waktu_target = 2.0;
  double hasil = hitung(fungsi_turunan, waktu_target);
  
  std::cout.precision(3);
  std::cout << "hasilnya adalah: " << hasil << " blablabla / detik" << std::endl;
  return 0;
}

File: turunan/part_1/turunan_pertama.py
from typing import Callable


def fungsi_posisi(t: float) -> float:
    return t**3 - 6 * t**2 + 9 * t


def fungsi_turunan(t: float) -> float:
    return 3 * t**2 - 12 * t + 9


def hitung(fungsi_turunan: Callable[[float], float], waktu: float) -> float:
    return fungsi_turunan(waktu)

if __name__ == "__main__":
    target_waktu: float = 2.0
    hasil = hitung(fungsi_turunan, target_waktu)
    print(f"kecepatannya adalah = {int(hasil)} blabla/detik")
